worker/worker.go

```diff
@@ -445,6 +462,179 @@ func (w *Worker) processClaim(claim taskClaim) {
 	}
 }
 
+// superseding returns any superseding task, and a function to be called when
+// processed to resolve other superseded tasks.
+func (w *Worker) superseding(claim taskClaim) (taskClaim, func()) {
+	// Create monitor for any problems we run into
+	m := w.monitor.WithPrefix("superseding")
+
+	var payload map[string]interface{}
+	if json.Unmarshal(claim.Task.Payload, &payload) != nil {
+		// Do nothing, if there is an error, it'll show up later and logging will be
+		// more natural.
+		return claim, func() {}
+	}
+
+	// Take supersederUrl out of the payload, as it would break the payload
+	// validation done in TaskRun. We attempt to hide superseding from the rest
+	// of the worker implementation, so that it's only creating a hack here.
+	supersederURL, hasSupersederURL := payload["supersederUrl"].(string)
+	delete(payload, "supersederUrl")
+	var err error
+	claim.Task.Payload, err = json.Marshal(payload)
+	if err != nil {
+		panic(errors.Wrap(err, "failed to serialize data we know to be JSON"))
+	}
+
+	// Do nothing, if there is no supersederUrl
+	if !hasSupersederURL || supersederURL == "" {
+		return claim, func() {}
+	}
+
+	// Fetch list of superseding tasks from superseder
+	g := got.New()
+	r, err := g.Get(supersederURL + "?taskId=" + claim.Status.TaskID).Send()
+	if err != nil {
+		m.Warnf("failed to contact supersederUrl: '%s'", supersederURL)
+		return claim, func() {}
+	}
+
+	var result struct {
+		Supersedes []string `json:"supersedes"`
+	}
+	if json.Unmarshal(r.Body, &result) != nil {
+		m.Warnf("failed to JSON parse result from supersederUrl: '%s'", supersederURL)
+		return claim, func() {}
```

Comment:
We should report this to sentry. Does `Warnf` take care of that?

Reply:
Nope, and no... We don't do that in docker worker... we could print it to task log.. but at this point I'm okay with a less than ideal solution.

I want to rip it out later anyways.